pkg/user/users_test.go

```diff
@@ -0,0 +1,606 @@
+	expect := assert.New(t)
+	orgIds, err := service.ReadOrgIDs(ctx, false, 3, "")
+	if expect.NoError(err) {
+		expect.Equal(orgIds, knownOrgIDs)
+	}
+}
+
+func TestReadAllOrgIDs(t *testing.T) {
+	expect := assert.New(t)
+	orgIds, err := service.ReadAllOrgIDs(ctx)
+	if expect.NoError(err) {
+		expect.Equal(orgIds, knownOrgIDs)
+	}
+}
+
+func TestReadUsersByOrgID(t *testing.T) {
+	expect := assert.New(t)
+	users, err := service.ReadUsersByOrgID(ctx, orgID4, true, 2, "")
+	if expect.NoError(err) && expect.NotEmpty(users) {
+		expect.Subset(knownUsers, users)
+	}
+}
+
+func TestReadUsersByOrgIDAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	user, err := service.ReadUsersByOrgIDAsJSON(ctx, orgID3, true, 1, "")
+	if expect.NoError(err) && expect.NotEmpty(user) {
+		expect.Contains(string(user), u30.OrgID)
+	}
+}
+
+func TestReadAllUsersByOrgID(t *testing.T) {
+	expect := assert.New(t)
+	users, err := service.ReadAllUsersByOrgID(ctx, orgID1)
+	if expect.NoError(err) && expect.NotEmpty(users) {
+		var numOfUsers []int
+		var usersIds []string
+		for i, v := range users {
+			numOfUsers = append(numOfUsers, i)
+			usersIds = append(usersIds, v.OrgID)
+		}
+		expect.GreaterOrEqual(len(numOfUsers), 1)
+		expect.Subset(knownOrgIDs, usersIds)
+	}
+}
+
+func TestReadAllUsersByOrgIDNegative(t *testing.T) {
+	expect := assert.New(t)
+	noUsers, err := service.ReadAllUsersByOrgID(ctx, orgID2)
+	if expect.NoError(err) {
+		expect.Empty(noUsers)
+	}
+}
+
+func TestReadAllUsersByOrgIDAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	users, err := service.ReadAllUsersByOrgIDAsJSON(ctx, orgID4)
+	if expect.NoError(err) && expect.NotEmpty(users) {
+		expect.Contains(string(users), u40.OrgID)
+	}
+}
+
+//------------------------------------------------------------------------------
+// Users by Role
+//------------------------------------------------------------------------------
+
+func TestReadRoles(t *testing.T) {
+	expect := assert.New(t)
+	roles, err := service.ReadRoles(ctx, false, 2, "o")
+	if expect.NoError(err) {
+		expect.Subset(allRoles, roles)
+	}
+}
+
+func TestReadAllRoles(t *testing.T) {
+	expect := assert.New(t)
+	roles, err := service.ReadAllRoles(ctx)
+	if expect.NoError(err) {
+		expect.Equal(roles, allRoles)
+	}
+}
+
+func TestReadUsersByRole(t *testing.T) {
+	expect := assert.New(t)
+	usersByRole, err := service.ReadUsersByRole(ctx, "vp", false, 1, "")
+	if expect.NoError(err) && expect.NotEmpty(usersByRole) {
+		expect.Subset(knownUsers, usersByRole)
+	}
+}
+
+func TestReadUsersByRoleAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	usersByRole, err := service.ReadUsersByRoleAsJSON(ctx, "engineer", false, 1, "")
+	if expect.NoError(err) && expect.NotEmpty(usersByRole) {
+		expect.Contains(string(usersByRole), u20.Roles[0])
+	}
+}
+
+func TestReadAllUsersByRole(t *testing.T) {
+	expect := assert.New(t)
+	usersByRole, err := service.ReadAllUsersByRole(ctx, "assistant")
+	if expect.NoError(err) && expect.NotEmpty(usersByRole) {
```

Comment:
The CRUD user does not have the role "assistant", so we expect the number of users to be 4. This block could be simplified to testing that the length of `usersByRole` is 4.